marshmallow_jsonapi/schema.py

```diff
@@ -286,21 +286,36 @@ def format_errors(self, errors, many):
 
         formatted_errors = []
         if many:
-            for index, errors in errors.items():
-                for field_name, field_errors in errors.items():
-                    formatted_errors.extend(
-                        [
-                            self.format_error(field_name, message, index=index)
-                            for message in field_errors
-                        ]
-                    )
+            for index, i_errors in errors.items():
+                formatted_errors.extend(self._get_formatted_errors(i_errors, index))
         else:
-            for field_name, field_errors in errors.items():
-                formatted_errors.extend(
-                    [self.format_error(field_name, message) for message in field_errors]
-                )
+            formatted_errors.extend(self._get_formatted_errors(errors))
+
         return {"errors": formatted_errors}
 
+    def _get_formatted_errors(self, errors, index=None):
+        fmtd_errors = []
```

Comment:
Can you use `itertools.chain` and list comprehensions instead of all these for loops?

Reply:
No problem. Now it's a bit hard to read, but if you think it'll boost the performance..

Reply:
Tests are passing though